pkg/rulebindingmanager/cache/cache.go

```diff
@@ -84,19 +84,13 @@ func (c *RBCache) AddNotifier(n *chan rulebindingmanager.RuleBindingNotify) {
 
 // ------------------ watcher.Watcher methods -----------------------
 
-func (c *RBCache) AddHandler(ctx context.Context, obj *unstructured.Unstructured) {
+func (c *RBCache) AddHandler(ctx context.Context, obj runtime.Object) {
 	var rbs []rulebindingmanager.RuleBindingNotify
 
-	switch obj.GetKind() {
-	case "Pod":
-		pod, err := unstructuredToPod(obj)
-		if err != nil {
-			logger.L().Error("failed to convert unstructured to pod", helpers.Error(err))
-			return
-		}
+	if pod, ok := obj.(*corev1.Pod); ok {
```

Comment:
Do you want to keep the switch case which is a bit more elegant?
You can check the kind, see [here](https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#Object).

Reply:
I agree with the switch case, however I didn't find a way to use type inference which is what we want here